railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt

```diff
@@ -32,9 +32,14 @@ when "production"
   # If you are running more than 1 thread per process, the workers count
   # should be equal to the number of processors (CPU cores) in production.
   #
-  # Automatically detect the number of available processors in production.
+  # Automatically detect the number of available processors in production
+  # when WEB_CONCURRENCY is set to "auto".
   require "concurrent-ruby"
-  workers_count = Integer(ENV.fetch("WEB_CONCURRENCY") { Concurrent.available_processor_count })
+  workers_count = if ENV["WEB_CONCURRENCY"] == "auto"
```

Comment:
This doesn't actually work since puma itself attempts to parse WEB_CONCURRENCY from ENV. And finding auto, it errors out.

Reply:
Thanks for catching this, @dhh

Sorry for the oversight. I initially tested this approach on an older application with Puma 4.3.12, which has more CPUs available on the server, to see how `available_processor_count` would work. I upgraded `concurrent-ruby` gem, but forgot the Puma. 

I see now that starting from Puma 5+, it uses `WEB_CONCURRENCY` directly and doesn't support `auto`. 

I should have tested compatibility with the latest stack.

I'll open a PR to update `tuning_performance_for_deployment.md`.

Thanks for the fixes.

Reply:
@dhh, @rafaelfranca 

I opened PR https://github.com/rails/rails/pull/52545 to update the docs.